src/main/java/com/server/ResponseBuilder.java

```diff
@@ -13,10 +14,21 @@
         this.responseHeaderBuilder = new ResponseHeaderBuilder(requestRouter);
     }
 
-    public String getResponse(RequestParams requestParams) throws IOException {
-        String body = this.responseBodyBuilder.getBody(requestParams, responseParams);
-        String header = this.responseHeaderBuilder.getHeader(requestParams, this.responseBodyBuilder.responseParams);
+    public byte[] getResponse(RequestParams requestParams) throws IOException {
+        byte[] body = this.responseBodyBuilder.getBody(requestParams, responseParams);
+        byte[] header = this.responseHeaderBuilder.getHeader(requestParams, this.responseBodyBuilder.responseParams);
 
-        return body == null ? header : header + body;
+        if (body.length == 0) {
+            return header;
+        }
+        else {
```

Comment:
Is this code block responsible for concatenating the two byte arrays? What are your thoughts on extracting it into a pure function?